app/src/main/java/edu/uc/kovaciad/slicetracker/dao/IMaterialDAO.kt

```diff
@@ -14,9 +14,21 @@ interface IMaterialDAO {
     fun findByName(name: String): Material
 
     @Insert
-    fun insertMaterial(material: Material)
-
-    @Delete
-    fun deleteMaterial(material: Material)
+    suspend fun insertMaterial(material: Material)
+    /**
+     * When updating a row with a value already set in a column,
+     * replaces the old value with the new one.
+     *
+     * @param  material new value to write
+     */
+    @Update
+    suspend fun update(material: Material)
+    /**
+     * Deletes all values from the table.
+     *
+     * This does not delete the table, only its contents.
+     */
+    @Query("DELETE FROM material")
+    suspend fun clear()
```

Comment:
I added a update function because when updating a row with a value already set in a column this replaces the old value with the new one. Also, I modified your delete function. It deletes all values from the table. This does not delete the table, only its contents.
Check out: https://developer.android.com/reference/androidx/room/Update
https://developer.android.com/reference/kotlin/androidx/room/Delete